micrometer-core/src/main/java/io/micrometer/core/instrument/binder/cache/CaffeineCacheMetrics.java

```diff
@@ -58,6 +62,10 @@ public class CaffeineCacheMetrics<K, V, C extends Cache<K, V>> extends CacheMete
      */
     public CaffeineCacheMetrics(C cache, String cacheName, Iterable<Tag> tags) {
         super(cache, cacheName, tags);
+
+        if (!cache.policy().isRecordingStats()) {
+            log.warn("The cache '{}' is not recording statistics.", cacheName);
```

Comment:
I think we should be more explicit about the consequences of this. Does it mean metrics will never be recorded unless `Caffeine#recordStats` is called and metrics are re-bound?

Reply:
@shakuzen Their values will be zero without it being called. I tried to refine the warning in d43ddb3.